packages/sdk/src/client.ts

```diff
@@ -363,6 +374,118 @@ export class AcrossClient {
     }
   }
 
+  /**
+   * Execute a swap quote by:
+   * 1. Executing approval transactions if present
+   * 2. Executing the swap/bridge transaction
+   * 3. Waiting for the deposit transaction to be confirmed
+   * 4. Waiting for the fill transaction to be confirmed
+   *
+   * See {@link executeSwapQuote} for more details.
+   *
+   * @example
+   * ```ts
+   * const quote = await client.getSwapQuote({
+   *   originChainId: 1,
+   *   destinationChainId: 10,
+   *   inputToken: "0x...",
+   *   outputToken: "0x...",
+   *   amount: "10",
+   *   depositor: "0x...",
+   * });
+   * const { depositId, swapTxReceipt, fillTxReceipt } = await client.executeSwapQuote({ swapQuote: quote });
+   * ```
+   *
+   * @param params - See {@link ExecuteSwapQuoteParams}.
+   * @returns The deposit ID and receipts for the deposit and fill transactions.
+   */
+  async executeSwapQuote(
+    params: MakeOptional<
+      ExecuteSwapQuoteParams,
+      | "logger"
+      | "originClient"
+      | "destinationClient"
+      | "integratorId"
+      | "walletClient"
+    >,
+  ): Promise<ExecuteSwapQuoteResponseParams> {
+    const logger = params?.logger ?? this.logger;
+    const originChainId = params.swapQuote.inputToken.chainId;
+    const destinationChainId = params.swapQuote.outputToken.chainId;
+    const originClient =
+      params?.originClient ?? this.getPublicClient(originChainId);
+    const destinationClient =
+      params?.destinationClient ?? this.getPublicClient(destinationChainId);
+    const integratorId = params?.integratorId ?? this.integratorId;
+    const walletClient = params?.walletClient ?? this?.walletClient;
+    const destinationSpokePoolAddress =
+      await this.getSpokePoolAddress(destinationChainId);
+
+    if (!walletClient) {
+      throw new ConfigError(
+        "WalletClient needs to be set to call 'executeSwapQuote'",
+      );
+    }
+
+    try {
+      return await executeSwapQuote({
+        ...params,
+        integratorId,
+        logger,
+        walletClient,
+        originClient,
+        destinationClient,
+        destinationSpokePoolAddress,
+      });
+    } catch (e) {
+      if (
+        !this.isTenderlyEnabled ||
+        !this.tenderly?.simOnError ||
+        !(
+          e instanceof AcrossApiSimulationError ||
+          e instanceof ContractFunctionExecutionError
+        )
+      ) {
+        throw e;
+      }
+
+      // The Across API only throws an `AcrossApiSimulationError` when simulating fills
+      // on the destination chain.
+      const isFillSimulationError = e instanceof AcrossApiSimulationError;
```

Comment:
Hmm, we need some adjustments here... Because this assumption is not true anymore. Or at least for the /swap endpoint. We also do swapTx simulations in the API which also might throw this 🤔